src/Tizen.Applications.Common/Tizen.Applications/EventSystem.cs

```diff
@@ -0,0 +1,172 @@
+        private readonly string _eventName = null;
+        private Interop.EventSystem.EventCallback _eventCallBack;
+
+        /// <summary>
+        /// Initializes the instance of the Event class.
+        /// </summary>
+        /// <param name="eventName">The name of the event.</param>
+        /// <exception cref="System.ArgumentException">Thrown when eventName is null or empty.</exception>
+        /// <example>
+        /// <code>
+        /// EventSystem ev = new EventSystem("tizen.system.event.screen_autorotate_state");
+        /// </code>
+        /// </example>
+        /// <since_tizen> 6 </since_tizen>
+        public EventSystem(string eventName)
+        {
+            if (String.IsNullOrEmpty(eventName))
+            {
+                throw new InvalidOperationException("null event name");
+            }
+            _eventName = eventName;
+        }
+
+        /// <summary>
+        /// Destructor of the Event class.
+        /// </summary>
+        ~EventSystem()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Called when a event is received.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// EventSystem ev = new EventSystem("tizen.system.event.screen_autorotate_state");
+        /// ev.EventReceived += EventReceivedCallback;
+        /// static void EventReceivedCallback(EventSystemArgs e)
+        /// {
+        ///     Console.WriteLine("Event Received ");
+        /// }
+        /// </code>
+        /// </example>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<EventSystemArgs> EventReceived;
+
+        /// <summary>
+        /// The name of the event.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public string EventName
+        {
+            get
+            {
+                return _eventName;
+            }
+        }
+        
+        /// <summary>
+        /// If true, the event is listening, otherwise false.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public bool Listening
+        {
+            get
+            {
+                return _listening;
+            }
+        }
+
+        /// <summary>
+        /// Listening event.
+        /// </summary>
+        /// <exception cref="System.InvalidOperationException">Thrown when portName is already used, when there is an I/O error.</exception>
+        /// <exception cref="System.ArgumentException">Thrown when there is an invalid parameter.</exception>
+        /// <exception cref="System.OutOfMemoryException">Thrown when out of memory.</exception>
+        /// <example>
+        /// <code>
+        /// EventSystem ev = new EventSystem("tizen.system.event.screen_autorotate_state");
+        /// ev.EventReceived += EventReceivedCallback;
+        /// ev.Listen();
+        /// </code>
+        /// </example>
+        /// <since_tizen> 6 </since_tizen>
+        public void Listen()
```

Comment:
Is this API really needed? As an app developer, calling Listen() after adding event handler is duplicated job, I think.